components/layout_2020/style_ext.rs

```diff
@@ -165,6 +173,82 @@ impl ComputedValuesExt for ComputedValues {
             self.writing_mode,
         )
     }
+
+    /// Returns true if this style has a transform, or perspective property set.
+    fn has_transform_or_perspective(&self) -> bool {
+        !self.get_box().transform.0.is_empty() || self.get_box().perspective != Perspective::None
+    }
+
+    /// Get the effective z-index of this fragment. Z-indices only apply to positioned element
+    /// per CSS 2 9.9.1 (http://www.w3.org/TR/CSS2/visuren.html#z-index), so this value may differ
+    /// from the value specified in the style.
+    fn effective_z_index(&self) -> i32 {
+        match self.get_box().position {
+            ComputedPosition::Static => {},
+            _ => return self.get_position().z_index.integer_or(0),
+        }
+
+        0
```

Comment:
nit: would probably be simpler and more idiomatic to simply write:

    match self.get_box().position {
        ComputedPosition::Static => 0,
        _ => self.get_position().z_index.integer_or(0),
    }

or even

    if self.get_box().position == ComputedPosition::Static {
        0
    } else {
        self.get_position().z_index.integer_or(0)
    }

Reply:
Fixed!